### PR TITLE
fix: JSONForm Phone number field required validation

### LINKED ISSUE #13174: [Bug]: Validation for isRequired in Phone Number Input type is not working properly in Json form
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

[LOOM DEMO](https://www.loom.com/share/f1c32534b541491a940455439785320b)

### Steps To Reproduce

1. Add Json form onto canvas
2. Make a field of Phone number input type
3. Enable the field as Required 
4. Now add number input and observe the error that says input is invalid

### Public Sample App

_No response_

### Version

Cloud/Production

### PR DESCRIPTION
**Pull request by ashit-rath**
## Description

Required validation for JSONForm phone number field did not work correctly. It sort of worked in the opposite way i.e when enter a value it threw an error and on empty the error went away.

Reason
The validation logic was incorrect and tests were there to check this but the jest test was testing another function present in different file than the Phone number field 🤦‍♂️

Fixes #13174

## Type of change

- Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?

- Jest Test

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

## Test coverage results :test_tube:
<details><summary>:green_circle: Total coverage has increased</summary>

    // Code coverage diff between base branch:release and head branch: fix/13174-jsonform-phone-validation 
Status | File | % Stmts | % Branch | % Funcs | % Lines 
 -----|-----|---------|----------|---------|------ 
 :green_circle: | total | 56.58 **(0.02)** | 38.37 **(0.04)** | 35.85 **(0.01)** | 56.83 **(0.03)**
 :red_circle: | app/client/src/components/propertyControls/JSONFormComputeControl.tsx | 24.47 **(-0.79)** | 9.76 **(0)** | 11.76 **(0)** | 24.1 **(-0.9)**
 :green_circle: | app/client/src/widgets/JSONFormWidget/fields/PhoneInputField.tsx | 46 **(18)** | 55 **(55)** | 20 **(20)** | 48.89 **(20)**
 :green_circle: | app/client/src/widgets/JSONFormWidget/widget/index.tsx | 28.7 **(1.81)** | 0 **(0)** | 20.69 **(0.69)** | 29.13 **(1.94)**
 :green_circle: | app/client/src/widgets/JSONFormWidget/widget/propertyConfig/helper.ts | 98.18 **(9.12)** | 79.17 **(6.09)** | 92.31 **(6.6)** | 98 **(10.07)**</details>

### CODE REVIEW

File: app/client/src/widgets/JSONFormWidget/fields/PhoneInputField.tsx [hunk 1]
```diff
@@ -61,7 +61,7 @@ export const isValid = (
 
   const parsedRegex = parseRegex(schemaItem.regex);
 
-  return parsedRegex ? parsedRegex.test(inputValue) : isEmptyValue;
+  return parsedRegex ? parsedRegex.test(inputValue) : true;
```
Comment (sbalaji1192) [inline app/client/src/widgets/JSONFormWidget/fields/PhoneInputField.tsx:64]: suggestion:
```return !parsedRegex || parsedRegex.test(inputValue)```

**Review by sbalaji1192 — DISMISSED** [review]

**Comment by chandannkumar** [comment]
Tested this PR and working as Expected

**Pull request merged**